connexion/lifecycle.py

```diff
@@ -34,4 +34,21 @@ def __init__(self,
         self.mimetype = mimetype
         self.content_type = content_type
         self.body = body
-        self.headers = headers or {}
+        self.headers = ConnexionHeaders(headers)
+
+
+class ConnexionHeaders(object):
```

Comment:
How long has this API existed?  Is it worth making it usable as a map/dict for eternity?  Was this API documented or was it internal?

I guess I'm having a hard time understanding why a shim API that's in the process of being developed needs backwards compatibility.  If I insta-crash on anyone accessing `ConnexionHeaders`, it only breaks a handful of unit tests, and those only because they call `problem()`!

FWIW, I'd have an easier time understanding that someone could pass in a map/dict as headers, and that the constructor of ConnexionHeaders should migrate that into a better representation.